overlord/snapshotstate/backend/backend.go

```diff
@@ -378,7 +387,54 @@ func Save(ctx context.Context, id uint64, si *snap.Info, cfg map[string]interfac
 
 var isTesting = snapdenv.Testing()
 
-func addDirToZip(ctx context.Context, snapshot *client.Snapshot, w *zip.Writer, username string, entry, dir string) error {
+type addDirToZipOptions struct {
+	ExcludePaths []string `yaml:"exclude"`
+}
+
+func readSnapshotYaml(si *snap.Info, opts *addDirToZipOptions) error {
```

Comment:
As Miguel already pointed out, I think `func readSnapshotYaml(si *snap.Info) (*addDirToZipOptions, error) {` is more idiomatic go